File: admin-frontend/src/schemas/admin-api-key.ts
import { CreatedUpdatedAt, UUIDSchema } from './generics';

export interface AdminAPIKeyCreate {
  name: string;
}

export interface AdminAPIKeyBase extends UUIDSchema, CreatedUpdatedAt {
  name: string;
  workspace_id: string;
}

export interface AdminAPIKeyCreateResponse extends AdminAPIKeyBase {
  token: string;
}

export interface AdminAPIKey extends AdminAPIKeyBase {
  token: '**********';
}

File: backend/fief/schemas/admin_api_key.py
from pydantic import UUID4, SecretStr

from fief.schemas.generics import BaseModel, CreatedUpdatedAt, UUIDSchema


class AdminAPIKeyCreate(BaseModel):
    name: str


class BaseAdminApiKey(UUIDSchema, CreatedUpdatedAt):
    name: str
    workspace_id: UUID4


class AdminAPIKeyCreateResponse(BaseAdminApiKey):
    token: str


class AdminAPIKey(BaseAdminApiKey):
    token: SecretStr
